modules/alookup/alookup_test.go

```diff
@@ -23,36 +23,41 @@ import (
 	"github.com/zmap/zdns/modules/miekg"
 )
 
+var mockResults = make(map[string]miekg.Result)
+
+var errCode = zdns.STATUS_NOERROR
+
 // Mock the actual Miekg lookup.
 func (s *Lookup) DoMiekgLookup(question miekg.Question, nameServer string) (interface{}, []interface{}, zdns.Status, error) {
 	if res, ok := mockResults[question.Name]; ok {
-		return res, nil, zdns.STATUS_NOERROR, nil
+		return res, nil, errCode, nil
 	} else {
-		return nil, nil, zdns.STATUS_NO_ANSWER, nil
+		return nil, nil, zdns.STATUS_NXDOMAIN, nil
 	}
 }
 
-var mockResults = make(map[string]miekg.Result)
-
-func TestDoLookup(t *testing.T) {
+func InitTest() (*zdns.GlobalConf, *GlobalLookupFactory, *RoutineLookupFactory, zdns.Lookup) {
+	mockResults = make(map[string]miekg.Result)
 	gc := new(zdns.GlobalConf)
 	gc.NameServers = []string{"127.0.0.1"}
 
 	glf := new(GlobalLookupFactory)
 	glf.GlobalConf = gc
-	glf.IPv4Lookup = true
-	glf.IPv6Lookup = false
 
 	rlf := new(RoutineLookupFactory)
 	rlf.Factory = glf
 	rlf.Client = new(dns.Client)
 
 	l, err := rlf.MakeLookup()
 	if l == nil || err != nil {
-		t.Error("Failed to initialize lookup")
+		panic("Failed to initialize lookup")
```

Comment:
@vishalmohanty I am assuming this change is to make it consistent with other lookup files?

Reply:
Thanks for catching this. I'm doing `t.Error` now.